src/Task/ConvertTranslatableTask.php

```diff
@@ -0,0 +1,268 @@
+        return $fields;
+    }
+
+    /**
+     * Gets all classes with FluentExtension
+     *
+     * @return array Array of classes to migrate
+     */
+    public function fluentClasses()
+    {
+        $classes = [];
+        $dataClasses = ClassInfo::subclassesFor(DataObject::class);
+        array_shift($dataClasses);
+        foreach ($dataClasses as $class) {
+            $base = DataObject::getSchema()->baseDataClass($class);
+            foreach (DataObject::get_extensions($base) as $extension) {
+                if (is_a($extension, FluentExtension::class, true)) {
+                    $classes[] = $base;
+                    break;
+                }
+            }
+        }
+        return array_unique($classes);
+    }
+
+    public function run($request)
+    {
+        // Extend time limit
+        set_time_limit(100000);
+
+        // we may need some privileges for this to work
+        // without this, running under sake is a problem
+        // maybe sake could take care of it ...
+        Security::setCurrentUser(
+            DefaultAdminService::singleton()->findOrCreateDefaultAdmin()
+        );
+
+        $this->checkInstalled();
+        $this->withTransaction(function (ConvertTranslatableTask $task) {
+            Versioned::set_stage(Versioned::DRAFT);
+            $classes = $task->fluentClasses();
+            $tables = DB::get_schema()->tableList();
+            if (empty($classes)) {
+                Debug::message('No classes have Fluent enabled, so skipping.', false);
+            }
+
+            foreach ($classes as $class) {
+                /** @var DataObject $class */
+
+                // Ensure that a translationgroup table exists for this class
+                $baseTable = DataObject::getSchema()->baseDataTable($class);
+                $groupTable = strtolower($baseTable . "_translationgroups");
+                if (isset($tables[$groupTable])) {
+                    $groupTable = $tables[$groupTable];
+                } else {
+                    Debug::message("Ignoring class without _translationgroups table $class", false);
+                    continue;
+                }
+
+                // Disable filter if it has been applied to the class
+                if (singleton($class)->hasMethod('has_extension')
+                    && $class::has_extension(FluentFilteredExtension::class)
+                ) {
+                    $class::remove_extension(FluentFilteredExtension::class);
+                }
+
+                // Select all instances of this class in the base table, where the Locale field is not null.
+                // Translatable has a Locale column on the base table in SS3, but Fluent doesn't use it. Newly
+                // created records via SS4 Fluent will not set this column, but will set it in {$baseTable}_Localised
+                $instances = DataObject::get($class, sprintf(
+                    '"%s"."Locale" IS NOT NULL',
```

Comment:
Although it's an _extreme_ edge case, assuming ANSI standard SQL isn't always a good thing (but it _should_ be!) - SQLServer uses `[` and `]` to delimit objects (not `"`). I believe there's a method or member field on the Database Adapter class that houses the more ideal value to use - perhaps we should leverage this as an example of good practice?

Reply:
We use ANSI as standard so that the DB connectors can convert as necessary into the appropriate syntax for the various DB drivers we support - at the moment SQL Server isn't one of those. This is mainly to provide compatibility for MySQL and PostgreSQL

Reply:
cool, fair enough :)